### DISCUSSION TITLE
TShock 4.5.14 for Terraria 1.4.3.4

### DISCUSSION THREAD

**Discussion by hakusaro** [Pryaxis/TShock#2574] [Show and tell]
This update adds preliminary support for Terraria 1.4.3.4, which is primarily a maintenance release that adds support for the Steam Deck!

Thank you to everyone who gave us feedback about Raspberry Pi and arm support. We [do not intend to discontinue support for arm](https://github.com/Pryaxis/TShock/discussions/2571), based on [your feedback](https://github.com/Pryaxis/TShock/discussions/2562).

## Notable changes
* Improved the `/grow` command to reduce code duplication, use `TileID` constants for less ambiguous types. (@drunderscore)
* Fixed item dupe via /logout & NPC. (@Terrarxxn)
* Added preliminary support for Terraria 1.4.3.4. Note that this has the side-effect of adding `IEntitySource` as the first parameter to `Item.NewItem` and `NPC.NewNPC`, and in `TSAPI`, `NpcLootDropEventArgs` passes `IEntitySource` as `Source`. If you're updating a plugin, you can either make something that implements with `IEntitySource` or just use `new EntitySource_DebugCommand()` [like TShock does](https://github.com/Pryaxis/TShock/commit/1b96ed8992110c5bbcc2ef952cc98459ea194dee). (@SignatureBeef, @Patrikkk, @hakusaro)

## Upgrading
To upgrade TShock, replace all of the files in the location of your server installation with all of the files from the release zip we provide. Overwrite all files. No data will be lost.

<hr /><em>This discussion was created from the release <a href='https://github.com/Pryaxis/TShock/releases/tag/v4.5.14'>TShock 4.5.14 for Terraria 1.4.3.4</a>.</em>

**Comment by wwa171** [comment]
Thanks for the quick update

**Comment by adriantomic** [comment]
Thanks for the update, and thanks for keeping the Raspberry Pi supported. Is there somewhere I can donate to this project?

**Comment by sidjadon** [comment]
hey[ is terraria cross platform](https://gameinstants.com/is-terraria-cross-platform/) if not how we can be enable it

**Reply by punchready** [reply]
nice advertising
